wgpu-types/src/lib.rs

```diff
@@ -2109,6 +2037,467 @@ pub enum TextureFormat {
     },
 }
 
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for TextureFormat {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use serde::de::{self, Error, Unexpected};
+
+        struct TextureFormatVisitor;
+
+        impl<'de> de::Visitor<'de> for TextureFormatVisitor {
+            type Value = TextureFormat;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("Invalid TextureFormat")
```

Comment:
Small nit: https://docs.rs/serde/latest/serde/de/trait.Visitor.html#tymethod.expecting

```suggestion
                formatter.write_str("a valid texture format")
```